admin/admin-model.php

```diff
@@ -288,33 +295,69 @@ public function update_language( $args ) {
+	 *                            loaded...
+	 *   @type int    $rtl        1 if rtl language, 0 otherwise.
+	 *   @type int    $term_group Language order when displayed.
+	 *   @type int    $lang_id    Optional, ID of the language to modify. An empty value means the language is being
+	 *                            created.
+	 *   @type string $flag       Optional, country code, {@see settings/flags.php}.
 	 * }
-	 * @return mixed The serialized description array updated.
+	 * @return string The serialized description array updated.
 	 */
-	protected function update_language_description( $args ) {
-		$language_term = get_term( (int) $args['lang_id'] );
+	protected function update_language_description( array $args ) {
+		if ( ! empty( $args['lang_id'] ) ) {
+			$language_term = get_term( (int) $args['lang_id'] );
 
-		if ( $language_term instanceof WP_Term ) {
-			$old_description = maybe_unserialize( $language_term->description );
+			if ( $language_term instanceof WP_Term ) {
+				$old_description = maybe_unserialize( $language_term->description );
+			}
 		}
 
 		if ( empty( $old_description ) || ! is_array( $old_description ) ) {
 			$old_description = array();
 		}
 
-		return maybe_serialize(
-			array_merge(
-				$old_description,
-				array( 'locale' => $args['locale'], 'rtl' => (int) $args['rtl'], 'flag_code' => empty( $args['flag'] ) ? '' : $args['flag'] )
-			)
+		$new_data      = array(
+			'locale'    => $args['locale'],
+			'rtl'       => ! empty( $args['rtl'] ) ? 1 : 0,
+			'flag_code' => empty( $args['flag'] ) ? '' : $args['flag'],
 		);
+		$language_data = array_merge( $old_description, $new_data );
+
+		/**
+		 * Allow to add data to store for a language.
+		 *
+		 * @param mixed[] $language_data {
+		 *     Data that will be stored.
+		 *
+		 *     @type string $locale    WordPress locale.
+		 *     @type int    $rtl       1 if rtl language, 0 otherwise.
+		 *     @type string $flag_code Country code.
+		 * }
+		 * @param mixed[] $args          {
+		 *     Arguments used to create the language.
+		 *
+		 *     @type string $name       Language name (used only for display).
+		 *     @type string $slug       Language code (ideally 2-letters ISO 639-1 language code).
+		 *     @type string $locale     WordPress locale. If something wrong is used for the locale, the .mo files will
+		 *                              not be loaded...
+		 *     @type int    $rtl        1 if rtl language, 0 otherwise.
+		 *     @type int    $term_group Language order when displayed.
+		 *     @type int    $lang_id    Optional, ID of the language to modify. An empty value means the language is
+		 *                              being created.
+		 *     @type string $flag       Optional, country code, {@see settings/flags.php}.
+		 * }
+		 */
+		$language_data = apply_filters( 'pll_language_metas', $language_data, $args );
+		// Don't allow to overwrite `$locale`, `$rtl`, and `$flag_code`.
```

Comment:
I agree with you not to allow to overwrite these properties. But having theme in the filter could be pertubating. Could we do something like
```php
$language_data = apply_filters( 'pll_language_metas', array(), $args );
```

Reply:
That's what I did first lol, but thought I was over-complicating things.

Reply:
Done in [f3c72a8](https://github.com/polylang/polylang/pull/1201/commits/f3c72a86f125fdcfe952183a88f44803d6e6d8f2).